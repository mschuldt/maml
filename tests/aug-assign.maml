arduino = Arduino(desktop=True)

@block(once)
def test():
    x = 1
    print(x) # 1
    x *= 5
    print(x) # 5
    x /= -1
    print(x) # -5
    x -= -6
    print(x) # 1
    y = -1 - -3
    print(y) # 2
    print(-1 - -3 >= 18) # 0
    print(-1 - -3 <= 18) # 1
    die(1)

arduino.send(test)
