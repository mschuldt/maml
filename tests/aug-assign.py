arduino = Arduino(desktop=True)

@block
def test():
    x = 1
    print_i(x) # 1
    x *= 5
    print_i(x) # 5
    x /= -1
    print_i(x) # -5
    x -= -6
    print_i(x) # 1
    y = -1 - -3
    print_i(y) # 2
    print_i(-1 - -3 >= 18) # 0
    print_i(-1 - -3 <= 18) # 1
    die()

arduino.send(test)
