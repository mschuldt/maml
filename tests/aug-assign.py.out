Initializing avm...
Ready.

1
5
-5
1
2
0
1
bye.
